### PR TITLE
Simple FIT Digitizer

### PR DESCRIPTION
**Pull request by AllaMaevskaya**

### CODE REVIEW

**Comment by AllaMaevskaya** [comment]
Dear Sandro,

I have no idea how to solve this

Best regards
Alla

File: Detectors/FIT/base/include/FITBase/Digit.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+// or submit itself to any jurisdiction.
+
+#ifndef ALICEO2_FIT_DIGIT_H_
+#define ALICEO2_FIT_DIGIT_H_
+
+#include "CommonDataFormat/TimeStamp.h"
+#include <iosfwd>
+#include "Rtypes.h"
+
+#include <boost/serialization/base_object.hpp> // for base_object
```
Comment (sawenzel) [inline Detectors/FIT/base/include/FITBase/Digit.h]: this include is likely not needed. Can you please remove it?

File: Detectors/FIT/base/FITBaseLinkDef.h [hunk 2]
```diff
@@ -15,7 +15,7 @@
 #pragma link off all functions;
 
 #pragma link C++ class o2::fit::Digit+;
-#pragma link C++ class o2::fit::Hit+;
-
+#pragma link C++ class o2::fit::Geometry + ;
+//#pragma link C++ class o2::fit::Hit+;
```
Comment (sawenzel) [inline Detectors/FIT/base/FITBaseLinkDef.h]: This file is likely not needed (There is another file under 'base/src/' of the same kind).

File: Detectors/FIT/base/src/FITBaseLinkDef.h [hunk 3]
```diff
@@ -15,5 +15,6 @@
 #pragma link off all functions;
 
 #pragma link C++ class o2::fit::Geometry;
+#pragma link C++ class o2::fit::Digit;
```
Comment (sawenzel) [inline Detectors/FIT/base/src/FITBaseLinkDef.h]: Please add a `+` after `Digit` to optimize the streamer.

File: Detectors/FIT/simulation/include/FITSimulation/Detector.h [hunk 4]
```diff
@@ -122,7 +122,8 @@ class Detector : public o2::Base::DetImpl<Detector>
   std::vector<Double_t> mReflMet;
 
   /// Container for data points
-  std::vector<HitType>* mHits;
+  std::vector<HitType>* mHits = nullptr;
+  ;
```
Comment (sawenzel) [inline Detectors/FIT/simulation/include/FITSimulation/Detector.h]: please remove `;`

File: Detectors/FIT/simulation/include/FITSimulation/MCLabel.h [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  Int_t mCFD = -1;
+
+ public:
+  MCLabel() = default;
+  MCLabel(Int_t trackID, Int_t eventID, Int_t srcID, Int_t cfd)
+    : o2::MCCompLabel(trackID, eventID, srcID), mCFD(cfd) {}
+  Int_t getCFD() const { return mCFD; }
+};
+} // namespace fit
+} // namespace o2
```
Comment (sawenzel) [inline Detectors/FIT/simulation/include/FITSimulation/MCLabel.h]: If you want to store such labels inside MCTruthContainer you will also need to create a dictionary for `o2::fit::MCLabel` as well as `o2::MCTruthContainer<o2::fit::MCLabel>`.

File: Detectors/FIT/simulation/src/Digitizer.cxx [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    amp[ipmt] = 0;
+    cfd[ipmt] = 0;
+  }
+  for (auto& hit : *hits) {
+    // TODO: put timeframe counting/selection
+    // if (timeframe == mTimeFrameCurrent) {
+    // timeframe = Int_t((mEventTime + hit.GetTime())); // to be replaced with uncalibrated time
+    mcp = hit.GetDetectorID();
+    if ((mcp < 96 && hit.GetTime() > lowTimeA && hit.GetTime() < highTimeA) ||
+        (mcp > 95 && hit.GetTime() > lowTimeC && hit.GetTime() < highTimeC)) {
```
Comment (sawenzel) [inline Detectors/FIT/simulation/src/Digitizer.cxx]: what do these constants `95` etc mean?

File: macro/gphysi.dat [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+ *                                                                                                 *
+ *                                                                                                 *
+ *              Standard TPAR for this run are                                                     *
+ *                                                                                                 *
+ *  CUTGAM=  1.00 MeV  CUTELE=  1.00 MeV  CUTNEU=  1.00 MeV  CUTHAD=  1.00 MeV  CUTMUO=  1.00 MeV  *
+ *  BCUTE =  1.00 MeV  BCUTM =  1.00 MeV  DCUTE = 10.00 TeV  DCUTM = 10.00 TeV  PPCUTM=  2.04 MeV  *
+ *  IPAIR=  1.  ICOMP=  1.  IPHOT=  1.  IPFIS=  0.  IDRAY=  0.  IANNI=  1.  IBREM=  1.  IHADR=  1. *
+ *  IMUNU=  1.  IDCAY=  1.  ILOSS=  2.  IMULS=  1.  IRAYL=  1.  ILABS=  1.  ISYNC=  0.  ISTRA=  0. *
+ *                                                                                                 *
+ *                                                                                                 *
```
Comment (sawenzel) [inline macro/gphysi.dat]: What is this file supposed to do? (We have likely a different mechanism to set cuts and processes)

Comment (sawenzel) [reply macro/gphysi.dat]: (Maybe it was committed by error?)

File: sw/MODULES/ubuntu1604_x86-64/BASE/1.0 [hunk 8]
```diff
@@ -0,0 +1,5 @@
+#%Module1.0
```
Comment (sawenzel) [inline sw/MODULES/ubuntu1604_x86-64/BASE/1.0]: file committed by error?

**Comment by sawenzel** [comment]
This looks quite nice. Don't worry about the o2-dev-fairroot build. It is not required to pass.

Could you still address the comments I made?

File: sw/MODULES/ubuntu1604_x86-64/BASE/1.0 (hunk 8 above)
Comment (AllaMaevskaya) [reply sw/MODULES/ubuntu1604_x86-64/BASE/1.0]: thank you. I will fix all of them

File: Detectors/FIT/simulation/src/Digitizer.cxx (hunk 6 above)
Comment (AllaMaevskaya) [reply Detectors/FIT/simulation/src/Digitizer.cxx]: Hello
MCPs from 0-95 are from A side, 95-207 - C side. Pisition of T0A is 320cm, T0C  -80cm

File: macro/gphysi.dat (hunk 7 above)
Comment (AllaMaevskaya) [reply macro/gphysi.dat]: sure, will be removed

File: Detectors/FIT/base/include/FITBase/Digit.h (hunk 1 above)
Comment (AllaMaevskaya) [reply Detectors/FIT/base/include/FITBase/Digit.h]: Yes, last one is not need now.

File: macro/gphysi.dat (hunk 7 above)
Comment (AllaMaevskaya) [reply macro/gphysi.dat]: It is GEANT3 output, I have no idea where it comes from, is it not exist here 
https://github.com/AllaMaevskaya/AliceO2/tree/fit-dev/Detectors/FIT/

Comment (sawenzel) [reply macro/gphysi.dat]: It is here:
https://github.com/AllaMaevskaya/AliceO2/tree/fit-dev/macro

Comment (AllaMaevskaya) [reply macro/gphysi.dat]: it was pushed by mistake, sure

**Pull request closed without merging**